lib/ansible/module_utils/ec2.py

```diff
@@ -728,3 +728,54 @@ def compare_aws_tags(current_tags_dict, new_tags_dict, purge_tags=True):
             tag_key_value_pairs_to_set[key] = new_tags_dict[key]
 
     return tag_key_value_pairs_to_set, tag_keys_to_unset
+
+
+def manage_tags(resource, resource_id, new_tags, purge_tags, connection, check_mode):
```

Comment:
Is this new or duplicate from another module?

Reply:
Duplicated from ec2_eni in https://github.com/ansible/ansible/pull/54950, which was duplicated from ec2_instance.  "Rule of 3" / DRY suggests I should generalize it.

Reply:
We can perhaps try to get this in, then have a single refactoring task at some point to consolidate.